x-pack/legacy/plugins/siem/public/pages/detection_engine/rules/details/index.tsx

```diff
@@ -259,13 +267,25 @@ const RuleDetailsPageComponent: FC<PropsFromRedux> = ({
                     >
                       <EuiFlexGroup alignItems="center">
                         <EuiFlexItem grow={false}>
-                          <RuleSwitch
-                            id={rule?.id ?? '-1'}
-                            isDisabled={userHasNoPermissions(canUserCRUD)}
-                            enabled={rule?.enabled ?? false}
-                            optionLabel={i18n.ACTIVATE_RULE}
-                            onChange={handleOnChangeEnabledRule}
-                          />
+                          <EuiToolTip
+                            position="top"
+                            content={
+                              rule?.type === 'machine_learning' && !hasMlPermissions
```

Comment:
I had left that one and instead added the `RuleDetails` component to our list of tech debt since there is so much branching on `null` in that file. It's due for some TLC... 😅